### ISSUE TITLE
protoc-gen-go_cli: Support nested oneof fields

### ISSUE THREAD

**Issue by tmc** [googleapis/gapic-generator-go#344]
googleapis/googleapis/blob/master/google/cloud/bigquery/connection/v1beta1/connection.proto fails to generate sucessfully.

https://github.com/googleapis/googleapis/blob/master/google/cloud/bigquery/connection/v1beta1/connection.proto#L133
has a nested field of: https://github.com/googleapis/googleapis/blob/master/google/cloud/bigquery/connection/v1beta1/connection.proto#L231
which has this oneof:
https://github.com/googleapis/googleapis/blob/master/google/cloud/bigquery/connection/v1beta1/connection.proto#L248

This fails in the code gen, for example:

```go
var CreateConnectionInput connectionpb.CreateConnectionRequest

var CreateConnectionFromFile string

var CreateConnectionInputConnectionProperties string

var CreateConnectionInputConnectionPropertiesCloudSql connectionpb.Connection_CloudSql

var CreateConnectionInputConnectionPropertiesCloudSql string
```

(`CreateConnectionInputConnectionPropertiesCloudSql` appears twice).

**Comment by codyoss** [comment]
cc @noahdietz

**Comment by noahdietz** [comment]
Ok, I've narrowed it down to mishandling of an [enum field](https://github.com/googleapis/googleapis/blob/master/google/cloud/bigquery/connection/v1beta1/connection.proto#L293) in the nested message's oneof field (which is also a nested message).

**Comment by noahdietz** [comment]
After fixing the enum handling, there were several other issues related to the complex nature of the API. The request message has a nested message, that has a oneof field that's a message (thus nested) and that message having another nested message. The oneof complicates the name construction. That's why this is taking longer. I might have to put it down at some point.

**Comment by noahdietz** [comment]
I think I've got a fix. I was able to generate and compile the bigquery connection CLI (and gapic-showcase, which depends on this tool for its CLI). Thanks for bringing this up. I will need to do a deeper rewrite at some point because the way this is built is not robust. 

BTW @tmc we will be generating and releasing a `cloud.google.com/go/bigquery/connection/apiv1beta1` client library soon. I noticed [your project](https://github.com/tmc/mkbigqueryconn/blob/master/bqconn/connection_client.go) and pinged some folks about it. Keep an eye out. Thanks for giving these tools a try in the meantime though!
